proposals/0000-reduce-fragmentation.md

```diff
@@ -0,0 +1,837 @@
+Support declarative transitions that can be optimized on the native side and avoid the need for `Animated`. Support `transitionProperty`, `transitionDuration`, `transitionTimingFunction`, and `transitionDelay` styles (see [transition](https://developer.mozilla.org/en-US/docs/Web/CSS/transition)). Consider dispatching the corresponding W3C transition events too.
+
+-----
+
+# Environment API
+
+## document
+
+Support the `document` object for common patterns such as listening to document resize, listening to capture or bubble phase events for the entire application (e.g., "outside click" pattern), etc.
+
+### Properties
+
+- [ ] [`activeElement`](https://developer.mozilla.org/en-US/docs/Web/API/Document/activeElement).
+- [ ] [`visibilityState`](https://developer.mozilla.org/en-US/docs/Web/API/Document/visibilityState)
+
+### Events
+
+* [ ] `scroll`.
+* [ ] `visibilitychange`.
+
+## IntersectionObserver
+
+Support the [`IntersectionObserver`](https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserver) API for observing the intersection of target elements. This can also be used as a building block for performance tooling.
+
+## MutationObserver
+
+Support the latest [`MutationObserver`](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver) API for watching changes to the host node tree. This can also be used as a building block for performance tooling.
+
+## ResizeObserver
+
+Support the latest [`ResizeObserver`](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver) API for responding the changes in the size of elements. Note that this API no longer includes positional coordinates and is optimized for width/height information.
+
+-----
+
+# Components API (DOM subset)
+
+A logical extension of support more React DOM props is to support a subset of React DOM components directly.
+
+This would provide a built-in declarative API for cross-platform elements in React Native. All unknown tags default to `<span>` layout. All known-but-unimplemented block-level tags default to `<div>` layout. Most tags can be mapped to existing React Native components and props, in the reverse of the [mapping used by React Native for Web](https://github.com/necolas/react-native-web/blob/b2eb3ca03ffc4b8570591b6c2082c6e4edae0e7f/packages/react-native-web/src/modules/AccessibilityUtil/propsToAccessibilityComponent.js#L12-L30).
+
+## Sections
+
+Content sectioning elements allow you to organize the document content into logical pieces. Use the sectioning elements to create a broad outline for your page content, including header and footer navigation, and heading elements to identify sections of content.
+
+* [ ] `article` is equivalent to `View` with `display:block` layout and `role="article"`.
+* [ ] `aside`
+* [ ] `div` is equivalent to `View` with `display:block` layout.
+* [ ] `footer`.
+* [ ] `h1-6`.
+* [ ] `header`.
+* [ ] `main`.
+* [ ] `nav`.
+* [ ] `section` is equivalent to `View` with `display:block` layout and `role="section"`.
```

Comment:
These elements are trivial to shim in React Native, other than the `block` layout.

Reply:
In the long term, I'm not sure. I think Seb suggested they could be "just there" like they are in react-dom. But in the immediate term, we might want/need to import them explicitly and in a way that can be recognized as a component (e.g., `<strict.div>` or `<html.div>`), because we won't be able to make changes to `style` or event props in the react-dom components. This would mean that this API is also opt-in for web users.